core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -103,15 +105,15 @@ object RequestChannel extends Logging {
       if (apiRemoteCompleteTimeNanos < 0)
         apiRemoteCompleteTimeNanos = responseCompleteTimeNanos
 
-      def nanosToMs(nanos: Long) = math.max(TimeUnit.NANOSECONDS.toMillis(nanos), 0)
+      def nanosToMs(nanos: Long) = math.max(nanos, 0).toDouble / TimeUnit.MILLISECONDS.toNanos(1)
```

Comment:
I am not sure we really need nanosecond precision for logging, especially since clocks don't provide nanosecond level accuracy. Would it make sense to either format the entries going into the request log or convert to micros to log with microsecond level precision? Btw, that was the intention in the previous code, I had just forgotten.

Reply:
Since we round the value to the nearest integer for the other case, it seems like the simplest is to convert to micros first.